multinet/uploaders/newick.py

```diff
@@ -19,41 +19,45 @@
 def validate_newick(tree: List[newick.Node]) -> None:
     """Validate newick tree."""
     data_errors: List[Dict[str, Any]] = []
-    unique_keys: List[str] = []
-    duplicate_keys: List[str] = []
-    unique_edges: List[dict] = []
-    duplicate_edges: List[dict] = []
+    unique_keys: Set[str] = set()
+    duplicate_keys: Set[str] = set()
+    unique_edges: Set[FrozenSet[Tuple[str, object]]] = set()
+    duplicate_edges: Set[FrozenSet[Tuple[str, object]]] = set()
```

Comment:
What's a `FrozenSet`?

Reply:
It's an immutable and hashable type which can be constructed from iterables. Since dictionaries themselves can't be hashed (because they're mutable), a frozenset is required to maintain data integrity while getting the benefits of hashing.

Here's an example of the conversion:

```
>>> a = {"a": 1, "b": 2, "c": 3}
>>> frozenset(a.items())
frozenset({('b', 2), ('a', 1), ('c', 3)})

```

Reply:
Cool. Did not really know about these before, thanks!